packages/jupyter-ai/src/components/chat-settings.tsx

```diff
@@ -285,6 +274,147 @@ export function ChatSettings(props: ChatSettingsProps): JSX.Element {
     );
   }
 
+  let language_model_section;
+  if (server.lmProviders.providers.length > 0) {
+    language_model_section = (
+      <Box>
+        <Select
+          value={lmProvider?.registry ? lmProvider.id + ':*' : lmGlobalId}
+          label="Completion model"
+          onChange={e => {
+            const lmGid = e.target.value === 'null' ? null : e.target.value;
+            if (lmGid === null) {
+              setLmProvider(null);
+              return;
+            }
+
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            const nextLmProvider = getProvider(lmGid, server.lmProviders)!;
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            const nextLmLocalId = getModelLocalId(lmGid)!;
+
+            setLmProvider(nextLmProvider);
+            setChatHelpMarkdown(nextLmProvider?.help ?? null);
+            if (nextLmProvider.registry) {
+              setLmLocalId('');
+              setShowLmLocalId(true);
+            } else {
+              setLmLocalId(nextLmLocalId);
+              setShowLmLocalId(false);
+            }
+          }}
+          MenuProps={{ sx: { maxHeight: '50%', minHeight: 400 } }}
+        >
+          <MenuItem value="null">None</MenuItem>
+          {server.lmProviders.providers.map(lmp =>
+            lmp.models
+              .filter(lm => lmp.chat_models.includes(lm))
+              .map(lm => (
+                <MenuItem value={`${lmp.id}:${lm}`}>
+                  {lmp.name} :: {lm}
+                </MenuItem>
+              ))
+          )}
+        </Select>
+        {showLmLocalId && (
+          <TextField
+            label={lmProvider?.model_id_label || 'Local model ID'}
+            value={lmLocalId}
+            onChange={e => setLmLocalId(e.target.value)}
+            fullWidth
+          />
+        )}
+        {chatHelpMarkdown && (
+          <RendermimeMarkdown
+            rmRegistry={props.rmRegistry}
+            markdownStr={chatHelpMarkdown}
+            complete
+          />
+        )}
+        {lmGlobalId && (
+          <ModelFields
+            fields={lmProvider?.fields}
+            values={fields}
+            onChange={setFields}
+          />
+        )}
+      </Box>
+    );
+  } else {
+    language_model_section = <p>No language models available.</p>;
+  }
+
+  let inline_completion_section;
+  if (isCompleterEnabled) {
```

Comment:
`isCompleterEnabled` is already used down there to make the select disabled. I am not sure if we should include it here. I think more value would be in making it:

```suggestion
  if (anyCompletionModelsAvailable) {
```

where `anyCompletionModelsAvailable` = `lmp.completion_models.length != 0`